### PR TITLE
ENH: add per axis interpolation to DiscreteLp

### LINKED ISSUE #201: Add per-axis support to DiscreteLp
Interpolation and `as_midp` should be settable per axis. Related: #159, #174.

### PR DESCRIPTION
**Pull request by kohr-h**
Fixes #201.

### CODE REVIEW

**Comment by adler-j** [comment]
No tests?

**Comment by kohr-h** [comment]
I can add some basic tests, sure, but the interpolation is being extensively tested already.

**Comment by kohr-h** [comment]
Basic tests added. A said, we have some tests for the actual interpolation, this addition only concerns the interface.

File: odl/discr/lp_discr.py [hunk 1]
```diff
@@ -99,10 +102,23 @@ def __init__(self, fspace, partition, dspace, exponent=2.0,
...
-            raise ValueError("'{}' is not among the supported interpolation "
-                             "types {}.".format(interp, _SUPPORTED_INTERP))
+        try:
+            # Got single string
+            interp, interp_in = str(interp + '').lower(), interp
+            if interp not in _SUPPORTED_INTERP:
+                raise ValueError("interpolation type '{}' not understood."
+                                 "".format(interp_in))
+            self._interp = [interp] * partition.ndim
+        except TypeError:
```
Comment (adler-j) [inline odl/discr/lp_discr.py]: What typeerror are we catching here?

**Comment by adler-j** [comment]
Looks good

File: odl/discr/lp_discr.py (hunk 1 above)
Comment (kohr-h) [inline odl/discr/lp_discr.py]: We catch the `interp + ''` error if `interp` is not a string-like type.

**Pull request merged**